source/table_view.tsx

```diff
@@ -35,7 +35,7 @@ export class TableView extends React.Component<Properties> implements
     for(let i = 0; i < this.props.labels.length; ++i) {
       this.headerRefs[i] = null;
     }
-    this.props.model.connect(this.handleUpdate.bind(this));
+    this.props.model.connect(this.setState.bind(this));
```

Comment:
You went from ```() => {this.forceUpdate();}``` to adding a new method that wraps ```this.forceUpdate();``` to now using ```this.setState,bind(this);```. All of these are incorrect in one way or another. The correct solution is simply:

```this.props.model.connect(this.forceUpdate.bind(this));```

Reply:
That doesn't work. I wanted to do that but the code keeps crashing.

Reply:
Okay good point then, what happens specifically?

Reply:
Everything is great until changeValues is called.

There is a warning: Warning: forceUpdate(...): Expected the last optional `callback` argument to be a function. Instead received: [object Object].

The error messages is: Error: Invalid argument passed as callback. Expected a function. Instead received: [object Object]

Reply:
Okay that's a good catch, ```forceUpdate``` takes a callback parameter, so you will need to bind it to a null callback as follows:

```this.forceUpdate.bind(this, null);```

Reply:
That explains it. I'll fix it now.